api/src/main/java/org/commonjava/indy/conf/IndyLoggingConfig.java

```diff
@@ -0,0 +1,164 @@
+
+    private String openshiftBuildCommit;
+
+    private String openshiftBuildName;
+
+    private String openshiftBuildNamespace;
+
+    public IndyLoggingConfig()
+    {
+    }
+
+    @Override
+    public String getDefaultConfigFileName()
+    {
+        return new File( IndyConfigInfo.CONF_INCLUDES_DIR, "logging.conf" ).getPath();
+    }
+
+    @Override
+    public InputStream getDefaultConfig()
+    {
+        return Thread.currentThread().getContextClassLoader().getResourceAsStream( "default-logging.conf" );
+    }
+
+    @Override
+    public synchronized void parameter( final String name, final String value )
+            throws ConfigurationException
+    {
+        switch ( name )
+        {
+            case HOST_NAME_KEY:
+            {
+                this.hostname = value;
+                break;
+            }
+            case BUILD_COMMIT_KEY:
+            {
+                this.openshiftBuildCommit = value;
+                break;
+            }
+            case BUILD_NAME_KEY:
+            {
+                this.openshiftBuildName = value;
+                break;
+            }
+            case BUILD_NAMESPACE_KEY:
+            {
+                this.openshiftBuildNamespace = value;
+                break;
+            }
+            default:
+            {
+                throw new ConfigurationException(
+                        "Invalid value: '{}' for parameter: '{}', they are not accepted for section: '{}'.", value,
+                        name, SECTION_NAME );
+            }
+
+        }
+    }
+
+    public String getHostname()
+    {
+        return hostname == null ? System.getenv( HOST_NAME_KEY ) : hostname;
+    }
+
+    @ConfigName( HOST_NAME_KEY )
+    public void setHostname( String host )
+    {
+        this.hostname = host;
+    }
+
+    public String getOpenshiftBuildCommit()
+    {
+        return openshiftBuildCommit == null ? System.getenv( BUILD_COMMIT_KEY ) : openshiftBuildCommit;
+    }
+
+    @ConfigName( BUILD_COMMIT_KEY )
+    public void setOpenshiftBuildCommit( String buildCommit )
+    {
+        this.openshiftBuildCommit = buildCommit;
+    }
+
+    public String getOpenshiftBuildName()
+    {
+        return openshiftBuildName == null ? System.getenv( BUILD_NAME_KEY ) : openshiftBuildName;
+    }
+
+    @ConfigName( BUILD_NAME_KEY )
+    public void setOpenshiftBuildName( String buildName )
+    {
+        this.openshiftBuildName = buildName;
+    }
+
+    public String getOpenshiftBuildNamespace()
+    {
+        return openshiftBuildNamespace == null ? System.getenv( BUILD_NAMESPACE_KEY ) : openshiftBuildNamespace;
+    }
+
+    @ConfigName( BUILD_NAMESPACE_KEY )
+    public void setOpenshiftBuildNamespace( String buildNamespace )
+    {
+        this.openshiftBuildNamespace = buildNamespace;
+    }
+
+    public Map<String, String> getEnvars()
+    {
+        Map<String, String> result = new HashMap<>();
+        result.put( HOST_NAME_KEY, getHostname() );
```

Comment:
You don't need config the hostname. In fact, if you call Java API to get hostname in the pod, it will return the pod name. I tried that via command 'hostname'. I guess Java Runtime.getRuntime().exec("hostname") will return just the same. 
I 'd suggest run this once and keep it somewhere and other components can just use it.

Reply:
Based on https://projects.engineering.redhat.com/browse/NOS-1658, John hopes these fields are able to be configured. Logic is when property value is null from the config, indy will fetch system env value instead.
I use System.getenv to get as John's suggestion, but these kinds of envars seem to be registed in profile, like export HOSTNAME, so we need confirm whether the envars are exported in openshift. Or IOUtils.toString( Runtime.getRuntime().exec( HOST_NAME_KEY).getInputStream()) would be a prefer. I'll confirm under 1658, thanks.

Reply:
FWIW, I just logged into the production Indy pod and ran `hostname` but the command was not found. I did get an accurate result by reading `/etc/hostname` but under the circumstances I think reading from the envars is probably simpler, given what else we're reading from envars.

$HOSTNAME is always available on a linux shell, so we should be safe here. I'm less sure what would happen if we started Indy outside of a normal shell, but fortunately we do need a real shell in order to access other things in the indy.sh file.